src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2408,6 +2408,17 @@ bool LibraryCallKit::inline_vector_convert() {
   const TypeVect* src_type = TypeVect::make(elem_bt_from, num_elem_from, is_mask);
   const TypeVect* dst_type = TypeVect::make(elem_bt_to, num_elem_to, is_mask);
 
+  // Safety check to prevent casting if source mask is of type vector
+  // and destination mask of type predicate vector and vice-versa.
+  // From X86 standpoint, this case will only arise over KNL target,
+  // where certain masks (depending on the species) are either propagated
+  // through a vector or predicate register.
+  if (is_mask &&
+      ((src_type->isa_vect() && dst_type->isa_vectmask()) ||
```

Comment:
Unfortunately, this will prevent all vectmask cast operations, since currently a vectmask type is also a vect type. @jatin-bhateja

Reply:
I created JDK-8275025 with the simple fix, but I am not able to verify that on x86. Please have a check: https://github.com/openjdk/panama-vector/pull/151